core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -316,25 +429,36 @@ impl SchedulerController {
         let mut error_counts = TransactionErrorMetrics::default();
         for chunk in packets.chunks(CHUNK_SIZE) {
             let mut post_sanitization_count: usize = 0;
-            let (transactions, fee_budget_limits_vec): (Vec<_>, Vec<_>) = chunk
+
+            let mut arc_packets = Vec::with_capacity(chunk.len());
+            let mut transactions = Vec::with_capacity(chunk.len());
+            let mut fee_budget_limits_vec = Vec::with_capacity(chunk.len());
```

Comment:
doing these allocation outside the loop and clearing the containers here might be better  for perf

Reply:
Tried to allocate outside in the original PR but coudlnt' get it to work since we need ownership of the transactions (and others) later when we insert to the container.

...but now revisiting I can simply do `x.drain(..)` in the final interior loop and that should give us an iterator while preserving the allocation.
Will make the change here and then also submit a PR to master.

Reply:
558477065f9d71417899ed4bb92c34293d93a143